### PR TITLE
feat(iframe_api): Implement readyToClose event and hangup command

### PR DESCRIPTION
**Pull request by hristoterezov**

### CODE REVIEW

File: conference.js [hunk 1]
```diff
@@ -1827,5 +1773,37 @@ export default {
...
             room.sendApplicationLog(JSON.stringify({name, value}));
         }
+    },
+    /**
+     * Disconnect from the conference and optionally request user feedback.
+     * @param {boolean} [requestFeedback=false] if user feedback should be
+     * requested
+     */
+    hangup (requestFeedback = false) {
+        const errCallback = (err) => {
```
Comment (lyubomir) [inline conference.js]: Why not `function errCallback(err)`? The arrow function is still an anonymous function. And the whole construct of assigning an anonymous function to a local variable looks not as simple to me.

File: conference.js [hunk 2]
```diff
@@ -1827,5 +1773,37 @@ export default {
...
+            else {
+                console.error('Error occurred during hanging up: ', err);
+                return Promise.resolve();
+            }
+        };
+
+        const disconnect = () => {
+            return disconnectAndShowFeedback(requestFeedback)
+                        .catch(errCallback);
+        };
```
Comment (lyubomir) [inline conference.js]: 1. Why not `function disconnect()`?

2. The value of disconnect is not used outside the `if (!conferenceLeftListener)` so why define it outside the `if`?

File: conference.js [hunk 3]
```diff
@@ -1827,5 +1773,37 @@ export default {
...
+            // If we want to break out the chain in our error handler, it needs
+            // to return a rejected promise. In the case of feedback request
+            // in progress it's important to not redirect to the welcome page
+            // (see below maybeRedirectToWelcomePage call).
+            if (err === UIErrors.FEEDBACK_REQUEST_IN_PROGRESS) {
+                return Promise.reject('Feedback request in progress.');
+            }
+            else {
+                console.error('Error occurred during hanging up: ', err);
+                return Promise.resolve();
```
Comment (lyubomir) [inline conference.js]: Is anyone using these `Promise`s that are being returned? I skimmed to the differences for occurrences of `hangup` and didn't see anyone caring about the return value. If someone is using it, why is there no jsdoc about the return value?

File: conference.js (hunk 1 above)
Comment (hristoterezov) [inline conference.js]: This is not new code. I moved the hangup function to the conference object. But I'll fix it.

**Comment by hristoterezov** [comment]
@lyubomir I refactored a little the hangup functionality and I think your comments are addressed.

**Pull request merged**